src/credential/sd_jwt_vc.rs

```diff
@@ -0,0 +1,169 @@
+    }
+
+    /// Access the revealed SD-JWT disclosures as a JSON encoded UTF-8 string.
+    pub fn sd_jwt_disclosures_as_json_encoded_utf8_string(&self) -> Option<String> {
+        self.disclosures
+            .as_ref()
+            .map(|disclosures| serde_json::to_string(disclosures).unwrap())
+    }
+
+    /// The keypair identified in the credential for use in a verifiable presentation.
+    pub fn key_alias(&self) -> Option<KeyAlias> {
+        self.key_alias.clone()
+    }
+}
+
+#[uniffi::export]
+impl SdJwtVc {
+    pub(crate) fn to_compact_sd_jwt_bytes(&self) -> Vec<u8> {
+        self.sd_jwt.as_bytes().to_vec()
+    }
+}
+
+#[uniffi::export]
+pub fn decode_reveal_sd_jwt(input: String) -> Result<String, SdJwtVcInitError> {
+    let jwt: SdJwtBuf = SdJwtBuf::new(input).map_err(|_| SdJwtVcInitError::InvalidSdJwt)?;
+    let revealed_jwt: RevealedSdJwt<AnyClaims> = jwt
+        .decode_reveal_any()
+        .map_err(|_| SdJwtVcInitError::JwtDecoding)?;
+    let claims: &JWTClaims = revealed_jwt.claims();
+    serde_json::to_string(claims).map_err(|_| SdJwtVcInitError::Serialization)
+}
+
+#[uniffi::export]
+pub fn convert_to_json_string(base64_encoded_bytes: &[u8]) -> Option<String> {
+    String::from_utf8(BASE64_STANDARD_NO_PAD.decode(base64_encoded_bytes).ok()?).ok()
+}
+
+#[derive(Debug, uniffi::Error, thiserror::Error)]
+pub enum SdJwtVcInitError {
+    #[error("failed to decode string as an SD-JWT of the form <base64-encoded-header>.<base64-encoded-payload>.<base64-encoded-signature>")]
+    CompactSdJwtDecoding,
+    #[error("failed to decode claim 'vc' as a W3C VCDM v1 or v2 credential")]
+    CredentialClaimDecoding,
+    #[error("'vc' is missing from the SD-JWT claims")]
+    CredentialClaimMissing,
+    #[error("failed to encode the credential as a UTF-8 string")]
+    CredentialStringEncoding,
+    #[error("failed to decode SD-JWT bytes as UTF-8")]
+    SdJwtBytesDecoding,
+    #[error("failed to decode SD-JWT as a JWT")]
+    JwtDecoding,
+    #[error("failed to decode JWT header as base64-encoded JSON")]
+    HeaderDecoding,
+    #[error("failed to decode JWT payload as base64-encoded JSON")]
+    PayloadDecoding,
+    #[error("failed to extract concealed claims (disclosures) from SD-JWT")]
+    DisclosureExtraction,
+    #[error("failed to verify the integrity of the SD-JWT with the disclosed claims")]
+    DisclosureVerification,
+    #[error("failed to decode disclosures")]
+    DisclosureDecoding,
+    #[error("invalid SD-JWT")]
+    InvalidSdJwt,
+    #[error("serialization error")]
+    Serialization,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_decode_reveal_sd_jwt() {
+        // Example SD-JWT input (you should replace this with a real SD-JWT string for a proper test)
+        let sd_jwt_input = "eyJhbGciOiJFZERTQSIsImtpZCI6IjEifQ.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
```

Comment:
This is not an SD-JWT by the way, this should end with a `~`

Reply:
If you want an SD-JWT to put in the tests, you can grab one from [the spec](https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-08.html).

Reply:
Updated SD-JWT provided in the `tests/examples/sd_vc.jwt` file.